src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -0,0 +1,66 @@
+package me.arminb.sara.dao;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import me.arminb.sara.entities.User;
+import org.bson.Document;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository("userDAO")
+public class UserDAOImpl implements UserDAO {
+
+    @Autowired
+    private MongoDatabase database;
+    //private MongoTemplate mongoTemplate;
+
+    @Override
+    public void findAll() {
+        try {
+            MongoCollection<Document> collection = database.getCollection("users");
+            for (Document doc : collection.find()) {
+                System.out.println(doc.toJson());
+            }
+        } catch (Exception e) {
+        }
+    }
+
+    @Override
+    public void addUser(User user) {
+        try {
+            MongoCollection<Document> collection = database.getCollection("users");
+            Document doc = new Document("userName", user.getUserName()).append("userEmail", user.getEmail()).append("userPassword",user.getPassword());
+            collection.insertOne(doc);
+        } catch (Exception e) {
```

Comment:
It is good to let the upper layer knows if everything went well or not. Also you should return the created User as a return value. The id field should be filled based on the insertion. So I would have something like this:
 public User addUser(User user) throws DataAccessException

where Data AccessException is a exception that you will create and means that something has gone wrong when inserting the record into the database.